gtsam/inference/BayesTree.h

```diff
@@ -27,6 +27,7 @@
 #include <gtsam/base/FastVector.h>
 
 #include <string>
+#include <queue>
```

Comment:
I would prefer that we add include headers and the code in the -inl.h file, rather than in this .h file.